### PR TITLE
[CALCITE] File annotations for DialectGenerate

### PR DESCRIPTION
**Pull request by AndrewPochapsky**
- Each extracted function and token assignment now has a comment above its declaration stating which file it was extracted from
- Added a new line separator between each extracted entity to improve readability
- Modified existing tests to work with annotations

### CODE REVIEW

File: buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java [hunk 1]
```diff
@@ -143,9 +143,15 @@ private void traverse(
       String fileName = f.getName();
       if (f.isFile() && fileName.endsWith(".ftl")) {
         try {
-          String fileText = new String(Files.readAllBytes(f.toPath()),
+          Path absoluteFilePath = f.toPath();
```
Comment (dasch-1) [inline buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java:146]: nit: This is getting to be a lot of contents in this try/catch block. Can we split out some things to not be in the try/catch?

File: buildSrc/subprojects/parser/src/test/resources/integrationTest/expected.ftl [hunk 2]
```diff
@@ -14,10 +14,14 @@
...
  * limitations under the License.
  */
+
+// Extracted from: parserTest/parserImpls.ftl
 <DEFAULT> TOKEN :
 {
 
 }
+
+// Extracted from: parserTest/intermediate/parserImpls.ftl
```
Comment (dasch-1) [inline buildSrc/subprojects/parser/src/test/resources/integrationTest/expected.ftl:24]: Should we have the parsingTest directory structure match the real one? e.g. parserTest/dialects/intermediate/parserImpls.ftl here?

File: buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java [hunk 3]
```diff
@@ -74,8 +74,10 @@
    * @param fileText The contents of the file to process
    * @param extractedData The object to which the parsed functions and token assignments
    *             will be added to
+   * @param filePath The path of the file being processed
    */
-  public void processFile(String fileText, ExtractedData extractedData) {
+  public void processFile(String fileText, ExtractedData extractedData,
+      String filePath) {
```
Comment (lisayin) [inline buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java:80]: Will there be other places that might call processFile without filePath that would still be a legitimate call?

File: buildSrc/subprojects/parser/src/test/resources/integrationTest/expected.ftl (hunk 2 above)
Comment (AndrewPochapsky) [reply buildSrc/subprojects/parser/src/test/resources/integrationTest/expected.ftl:24]: Good idea, will make the change.

File: buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java (hunk 3 above)
Comment (AndrewPochapsky) [reply buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java:80]: Not at the moment but I can see it possibly happening in the future. I'll make this parameter optional by having a null check before the annotation gets added.

File: buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java (hunk 1 above)
Comment (AndrewPochapsky) [reply buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java:146]: Will do

File: buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java [hunk 4]
```diff
@@ -168,31 +169,31 @@ private void assertFileProcessed(String testName) {
   }
 
   @Test public void processFileEmpty() {
-    assertFileProcessed("empty");
+    assertFileProcessed("empty", true);
```
Comment (dasch-1) [inline buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java:172]: nit: Use comment (here and elsewhere) to indicate the name of the parameter (e.g. /*specifyFilePath=*/)

**Pull request merged**